docs/proposals/post-deployment-hooks.md

```diff
@@ -0,0 +1,125 @@
+# Deployment Hooks
+
+## Abstract
+
+A proposal for a deployment code execution API integrated with the deployment lifecycle.
+
+
+## Motivation
+
+Deployment hooks are needed to provide users with a way to execute arbitrary commands necessary to complete a deployment, targeting executables which may be maintained outside the docker image being deployed.
+
+Goals of this design:
+
+1. Identify deployment hook use cases
+2. Define the integration of deployment hooks and the deployment lifecycle
+3. Describe a deployment hook API
+
+
+## Comparison of potential approaches
```

Comment:
I would make a note here saying that fundamentally, these are the choices you have to evaluate the use-cases in terms of, since this ties into the next section.